app/utilities.py

```diff
@@ -1,6 +1,28 @@
 from typing import Dict
+from app.data import MongoDB
 
+db = MongoDB("UnderdogDevs")
 
 def dict_to_str(data: Dict) -> str:
     """Convert dictionaries into easy to read strings."""
     return "\n" + "\n".join(f"{k}: {v}" for k, v in data.items())
+
+
+def financial_aid_gen(profile_id):
+    '''This is a function that queries the database
+    for variables attached to a mentees profile id and applies
+    that information into a forumla that returns financial aid
+    probability'''
+
+    try:
+        profile = db.read('Mentees', {"profile_id": profile_id})
+        E_l_dict = {"Beginner" : 0, "Intermediate" : 1, "Advanced" : 2, "Expert" : 3}
+        formerly_incarcerated = 1 if profile['result'][0]['formerly_incarcerated'] == 'true' else 0 
+        low_income = 1 if profile['result'][0]['low_income'] == 'true' else 0
+        experience_level = E_l_dict[profile['result'][0]['experience_level']]
+
+        return (((formerly_incarcerated/2 + low_income + pow(9.9*(experience_level), -1) - 0.025) / 1.577))
+        
+    except:
```

Comment:
You might consider removing the try/except as it's not really necessary as python will naturally throw an error

Reply:
Will python return a logical human-readable error, or will it just stop?
We did the try/except for returning a human-readable error in the case that the mentee id (or any of the formula's variables) was not in the database.

Reply:
Python will throw an ```IndexError```, which is not always the most intuitive admittedly; however, (this is where the refactor can make things easier) an option is to let the API throw the error instead of the actual function because the API will mainly be how the users interact with the function.